### PR TITLE
snapstate: fix update boot revisions for classic+modes devices

### PR DESCRIPTION
**Pull request by mvo5**
The existing `UpdateBootRevisions()` does not take into account that not all systems use the base as a boot participant. On a classic+modes system only the kernel is relevant. This PR updates this code to skip non-boot participants when checking if there were rollbacks across reboots.

### CODE REVIEW

**Review by pedronis — APPROVED** [review]
lgtm

File: tests/nested/manual/fde-on-classic/task.yaml [hunk 1]
```diff
@@ -111,4 +111,25 @@ execute: |
...
+  # precondition, the core22 is not a local version
+  remote.exec snap list core22 | NOMATCH " x1 "
+  # create modified core22
+  unsquashfs -d core22 "$CACHE_D"/core22_*.snap
+  touch core22/empty-file
+  snap pack --filename=core22-new.snap core22
+  remote.push "core22-new.snap"
+  # install and validate its now a local version
+  remote.exec sudo snap install --dangerous "core22-new.snap"
+  remote.exec sudo snap list core22 | MATCH " x1 "
```
Comment (alfonsosanchezbeato) [inline tests/nested/manual/fde-on-classic/task.yaml:126]: Shouldn't this tinker with modeenv so it has the old version of the base in `modeenv`? Or actually, what is happening is that there is no base in modeenv so `boot.GetCurrentBoot` was returning wrong information? What am I missing here?

**Comment by pedronis** [comment]
@alfonsosanchezbeato @mvo5 we also need separately to explore not setting this in the modeenv

**Review by alfonsosanchezbeato — COMMENTED** [review]
Looks good, but I'm not sure how is this triggered, how is that modeenv gets out of sync with the current base (see also questio below)?

**Comment by pedronis** [comment]
> Looks good, but I'm not sure how is this triggered, how is that modeenv gets out of sync with the current base

AFAIU the install code for classic with modes sets the base in the modeenv (I think it's the makebootable code?), but given that the base is not the boot base the code in the boot package that updates modeenv doesn't get triggered. The latter is fine, the first part is probably wrong/confusing.

**Comment by alfonsosanchezbeato** [comment]
> > Looks good, but I'm not sure how is this triggered, how is that modeenv gets out of sync with the current base
> 
> AFAIU the install code for classic with modes sets the base in the modeenv (I think it's the makebootable code?), but given that the base is not the boot base the code in the boot package that updates modeenv doesn't get triggered. The latter is fine, the first part is probably wrong/confusing.

Thanks, makes sense. Approving. @mvo5 I guess maybe a TODO in `makeRunnableSystem()` would be good to have?

**Review by alfonsosanchezbeato — APPROVED** [review]

**Pull request merged**